apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -142,14 +154,43 @@ export class ParseEventRequest {
 
       if (!subscriberIdExists) {
         throw new ApiException(
-          'subscriberId under property to is not configured, please make sure all the subscriber contains subscriberId property'
+          'subscriberId under property to is not configured, please make sure all subscribers contains subscriberId property'
         );
       }
     }
 
     return true;
   }
 
+  @Instrument()
+  private validateTriggerContext(
+    command: ParseEventRequestCommand,
+    reservedVariablesTypes: TriggerContextTypeEnum[]
+  ): void {
+    const invalidKeys: string[] = [];
+
+    for (const reservedVariableType of reservedVariablesTypes) {
+      const payload = command[reservedVariableType];
+      if (!payload) {
+        invalidKeys.push(`${reservedVariableType} object`);
+      } else {
+        const requiredVariablesContextNames = ReservedVariablesMap[reservedVariableType].map(
+          (variable) => variable.name
+        );
+        for (const variableName of requiredVariablesContextNames) {
+          const variableNameExists = typeof payload === 'string' ? payload : payload[variableName];
+          if (!variableNameExists) {
+            invalidKeys.push(`${variableName} property of ${reservedVariableType}`);
+          }
```

Comment:
On trigger, if missing the tenant object in trigger or missing the mandatory property 'identifier' will throw error